src/fem/fe_system.cc

```diff
@@ -0,0 +1,416 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ * 
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ * 
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ * 
+ * @file    fe_system.cc
+ * @brief   Class FESystem for compound finite elements.
+ * @author  Jan Stebel
+ */
+
+#include "fem/fe_system.hh"
+#include "system/global_defs.h"
+#include "quadrature/quadrature.hh"
+
+using namespace std;
+
+
+
+// template<unsigned int dim,unsigned int spacedim>
+// std::vector<FiniteElement<dim,spacedim> > expand_fe_vector(const std::vector<std::pair<FiniteElement<dim,spacedim>, unsigned int> > &fe)
+// {
+//   std::vector<FiniteElement<dim,spacedim> > finite_elements;
+//   
+//   for (auto fe_pair : fe)
+//     for (unsigned int i=0; i<fe_pair.second; i++)
+//       finite_elements.push_back(fe_pair.first);
+// 
+//   return finite_elements;
+// }
+
+
+
+template<unsigned int dim, unsigned int spacedim>
+FESystem<dim,spacedim>::FESystem(FiniteElement<dim,spacedim> *fe, unsigned int n)
+{
+  FiniteElement<dim,spacedim>::init(0, false);
+  std::shared_ptr<FiniteElement<dim,spacedim> > fe_ptr(fe);
+  fe_ = std::vector<std::shared_ptr<FiniteElement<dim,spacedim> > >(n, fe_ptr);
+  initialize();
+}
+
+
+
+template<unsigned int dim, unsigned int spacedim>
+void FESystem<dim,spacedim>::initialize()
+{
+  // indices of nodal dofs (for each node)
+  std::vector<unsigned int> node_basis[dim+1];
+  // indices of cell dofs
+  std::vector<unsigned int> cell_basis;
+  
+  unsigned int fe_index = 0;
+  unsigned int comp_offset = 0;
+  unsigned basis_offset = 0;
+  // for each base FE add components, support points, and other 
+  // information to the system
+  for (auto fe : fe_)
+  {
+    number_of_dofs += fe->n_dofs();
+    n_components_ += fe->n_components();
+
+    for (int i=0; i<=dim; ++i)
+    {
+      number_of_single_dofs[i] += fe->n_object_dofs(i, DOF_SINGLE);
+      number_of_pairs[i] += fe->n_object_dofs(i, DOF_PAIR);
+      number_of_triples[i] += fe->n_object_dofs(i, DOF_TRIPLE);
+      number_of_sextuples[i] += fe->n_object_dofs(i, DOF_SEXTUPLE);
+    }
+
+    for (int i=0; i<fe->get_generalized_support_points().size(); i++)
+        generalized_support_points.push_back(fe->get_generalized_support_points()[i]);
+
+    if (fe->polynomial_order() > order) order = fe->polynomial_order();
```

Comment:
Does it have a sense?

Reply:
Remove variable order. Only keep it as parameter of some constructors to construct particular order of element.

Reply:
OK